File: python/Find_the_Largest_Area_of_Square_Inside_Two_Rectangles_3047.py
class Solution:
    def largestSquareArea(self, bottomLeft: List[List[int]], topRight: List[List[int]]) -> int:
        maxArea = 0
        n = len(bottomLeft)
        for i in range(n):
            for j in range(i + 1, n):
                horSide = min(topRight[i][0], topRight[j][0]) - max(bottomLeft[i][0], bottomLeft[j][0])
                verSide = min(topRight[i][1], topRight[j][1]) - max(bottomLeft[i][1], bottomLeft[j][1])
                if horSide > 0 and verSide > 0:
                    maxArea = max(maxArea, min(horSide, verSide) ** 2)
        return maxArea

File: cpp/Find_the_Largest_Area_of_Square_Inside_Two_Rectangles_3047.cpp
class Solution {
public:
    long long largestSquareArea(vector<vector<int>>& bottomLeft, vector<vector<int>>& topRight) {
        long long maxArea = 0;
        int n = bottomLeft.size();
        for (int i = 0; i < n; i++) {
            for (int j = i + 1; j < n; j++) {
                int horSide = min(topRight[i][0], topRight[j][0]) - max(bottomLeft[i][0], bottomLeft[j][0]);
                int verSide = min(topRight[i][1], topRight[j][1]) - max(bottomLeft[i][1], bottomLeft[j][1]);
                int side = min(horSide, verSide);
                if (side <= 0) continue;
                maxArea = max(maxArea, side * 1LL * side);
            }
        }
        return maxArea;
    }
};
